packages/account-sdk/src/interface/public-utilities/spend-permission/methods/requestSpendPermission.ts

```diff
@@ -0,0 +1,78 @@
+import { SpendPermission } from ':core/rpc/coinbase_fetchSpendPermissions.js';
+
+import { ProviderInterface } from ':core/provider/interface.js';
+import { createSpendPermissionTypedData, toTimestampInSeconds } from '../utils.js';
+import { getHash } from './getHash.js';
+
+export type RequestSpendPermissionType = {
+  account: string;
+  spender: string;
+  token: string;
+  chainId: number;
+  allowance: bigint; // in wei
+  periodInDays: number;
+  start?: Date; // default to now
+  end?: Date; // default to never (start + 100 years)
+  salt?: string; // default to a random value by crypto.getRandomValues
+  extraData?: string; // default to '0x'
+};
+
+/**
+ * Requests user approval to create a new spend permission.
+ *
+ * This helper method opens a popup to ask the user to sign a spend permission,
+ * which allows a specified spender to spend tokens from the user's account up to
+ * a certain allowance within defined time periods.
```

Comment:
can we link out to docs from here? There's a lot of info to swallow in this docstring